extensions/oidc/runtime/src/main/java/io/quarkus/oidc/runtime/CodeAuthenticationMechanism.java

```diff
@@ -240,13 +241,19 @@ public Uni<SecurityIdentity> apply(TenantConfigContext tenantContext) {
 
                         });
             } else {
-                LOG.error(
-                        "Authentication has failed but no error handler is found, completing the code flow with HTTP status 401");
-                return Uni.createFrom().failure(new AuthenticationCompletionException());
+                final String errorMessage = """
+                        Authorization code flow has failed, error code: %s, error description: %s.
+                        Error handler path is not configured. Have a public JAX-RS resource listening
+                        on a path such as '/error' and point to it with 'quarkus.oidc.authentication.error-path=/error'.
+                        Completing the flow with HTTP status 401.
```

Comment:
It's quite big, logging is expensive and this is logger on error level. If I am correct, this will only be logged for when setup is wrong either on IDM or Quarkus or client side (or someone is using manual requests etc.). In that case it makes sense to log error code and description. But the error handler seems like useful information in DEV mode only. Valid token is required and incoming request doesn't have it, so it must be rejected. The error handler sounds like something regular user won't do, they cannot just change incoming credentials.

In short:

- do I read it right that this won't ever be logged for normal situations like when access token expired during request to UserInfo endpoint?
- is the mention about error handler useful to small amount of cases or do you really expect users to use it? I was alarmed at mention that Stephane requested it, because it would suggest you didn't hear often from users that they need it.

Reply:
@michalvavrik 

>It's quite big, logging is expensive and this is logger on error level. 

This covers an exception path  where the user is redirected back from Keycloak or other provider to Quarkus, but the user has failed to authenticate at the provider side for whatever reason. I wanted to have a very clear message how to intercept such flows (where a custom handler for ex returns a nice HTML page to the user explaining what happened). 
Compare it with the original message. It did not help Max in the slightest when he asked me what was going, what is the error handler, etc. I'm not worried at all abut the cost of logging in this case -:)

>do I read it right that this won't ever be logged for normal situations like when access token expired during request to UserInfo endpoint?

This specific one handles a moment where the authorization code flow is in progress and only the first part of it has completed, in this case, instead of returning `code` to Quarkus for it to be exchanged for tokens, the user is redirected with an error code.

>is the mention about error handler useful to small amount of cases or do you really expect users to use it? I was alarmed at mention that Stephane requested it, because it would suggest you didn't hear often from users that they need it.

I'm not sure what do you mean. Such errors are not happening often but it does not mean the handler is not needed. Steph asked for it a couple of years back, it was useful to have it added. But users, if they haven't done it, and start wondering in prod, why exactly the screen is empty now, they will see in the log what they need to do to make things better...

Reply:
@michalvavrik I had a similar comment from users re the case where the session cookie is > 4K, initially I logged something short, but there was an issue from users asking to make it clear in the logs what to do...

Reply:
I won't answer individual comments. I think you have explained your position. This is not attack on Stephan, it was my worry that you document edge feature in the logging message while users normally shouldn't use these loggers. I don't know what most users would do in that error handler, they cannot change that credentials are wrong, they can alter error message but then, they are giving away information in production.

Thank you for the explanation, let's leave it there.

Reply:
@michalvavrik Yeah, I know you meant to clarify things, np at all.

These error code and description, if any, they go via the browser, so these are standard original OAuth2 properties, so they are not sensitive properties as such. 
Now, the error handler can choose to return HTML page to the user, explaining with the error code alone that this or that part of the flow did not work out. 
Or it can just rethrow 401 but get hold of this error code and description and do something with it, like sending an alert or do something else...
Without the error handler - Quarkus app won't know what happened...

Reply:
> ... Without the error handler - Quarkus app won't know what happened...

I think being able to intercept error and propagate it in own manner makes sense. Or to produce standardized error code.

@sberyozkin I cannot close comment threads in your PRs (rights) :-) please consider it closed and thank you for the explanation.

Reply:
@michalvavrik Np at all, thanks too.